tests/engine/processes/calcjobs/test_calc_job.py

```diff
@@ -532,6 +532,23 @@ def test_get_hash(self, get_calcjob_builder):
         _, node = launch.run_get_node(builder)
         assert node.base.extras.get(node.base.caching._HASH_EXTRA_KEY) == node.get_hash()  # pylint: disable=protected-access
 
+    def test_get_objects_to_hash_core_version(self, get_calcjob_builder):
+        """Test that :meth:`aiida.orm.CalcJobNode._get_objects_to_hash` returns the expected objects."""
+        from importlib import import_module
+
+        builder = get_calcjob_builder()
+        _, node = launch.run_get_node(builder)
+        objects = node._get_objects_to_hash()
+        core_version = import_module(node.__module__.split('.', maxsplit=1)[0]).__version__
+
+        assert core_version not in objects
```

Comment:
Would this assertion fail before the change?? Shouldn't it be look at `obj['version']['core']` or something like it?